design/github-repository-restructuring.md

```diff
@@ -0,0 +1,52 @@
+# GitHub repository restructuring
+
+Currently, Strimzi has several Github repositories:
+* The _main_ Strimzi Kafka Operators repo
+* The Kafka Bridge repo
+* The Kafka OAuth repo
+* The Client Examples repo
+* The Strimzi Admin repo for the planned Strimzi Admin API
+* The Strimzi lab repo with some demos
+* The Strimzi website repo
+
+Despite us having multiple repositories, some things which are related to Strimzi it self seem to be only in the Operator repository.
+For example the design proposals or the governance documents are central and should apply to whole Strimzi, but are in the operator repository only.
+That might not be logical since some users might use only some parts and might miss these things.
+
+## Proposed changes
+
+### Governance
+
+We should create a new repo called `governance`.
+Into this repo, we should move the `GOVERNANCE.md` document with the Strimzi project governance as well as the `MAINTAINERS` file with the list of maintainers.
+In all other repositories, we should have a files `GOVERNANCE.md` and `MAINTAINERS.md` which will just contain a link to the file in the `governance` repository.
+Apart from these two files, the `governance` repo should also contain a license file and code of conduct file.
+
+All changes to the governance and maintainers should be done through PRs in that repository.
+The voting should be done either on the PR or on the mailing lists.
+
+### Design Proposals
+
+The design proposals are currently stored in the `design` folder of the `strimzi-kafka-operators` repository.
+Right now, the existing proposals focus on the operators.
+But it is expected that in the future we would have also proposals for the other components or proposals which include more than one repository.
+
+We should create a new repository called `proposals` which should be used for the proposals with changes for all our projects.
+The repository should contain an overview list of proposals with their state (Proposed, Approved, Implemented).
+For easier orientation, the proposals should be numbered starting from 0.
```

Comment:
Maintaining the overview list might be a bit burdensome, since it would require a PR to update the overview with a "proposed" when a proposal was created (which I assume is opening a PR with a new proposal, so two PRs). I assume the approval of a proposal is the same thing as merging its PR. When it's implemented we'd need another PR to update the overview again. I'm thinking it might be easier to use labels on the PRs themselves to track the status, avoiding the need for a dedicated overview. That doesn't really solve the numbering problem though. It might be a little awkward to use the PR number as the proposal number.

Or am I completely misunderstanding this?

Reply:
Well, I guess you opened multiple topics ... 
* I guess the state "Proposed" is indeed non-sense since that would be only in the PR and would need to be to changed to Approved before merging. 
* I thought that the statuses might be useful, But you are right that the effort is non-trivial.
* I do not think that keeping the status as PR label makes any sense. It is too hidden away to hard to find.
* The numbers in my mind were sequential not bound to PR numbers. I do not think the frequency of the proposals will be (at least initially) high enough to cause problema. And the number is important as it gives you an idea about the age of the proposal. Without the numbers, the directory with random set of MD files with some names would be hard to look at.
* I also think the overview file is useful as it can contain slightly linger name that the file names would etc.

So, I removed the state from the proposal, but for the time being pending other comments I kept the overview and the numbers.

Reply:
I agree to take proposal number and that having the status in the proposal needs too much work and PRs to keep them consistent but why not using the labels way suggested by Tom. I don't see it mentioned in the doc.

Reply:
Because I do not like the idea - I do not think the labels will work and anyone will be maintaining them and looking for them.

Reply:
And what are you proposing to highlight that a proposal was accepted?

Reply:
The accepted phase is simple ... it is merged when it is accepted. The implemented phase is complicated. I was proposing to maintain that status in the repository in the file. Tom was arguing that it would be too complicated since you would need a follow up PR. Which made sense, so I removed it.